src/transaction/log_replication.cpp

```diff
@@ -225,4 +276,41 @@ namespace cublog
 	m_parallel_replication_redo->wait_for_idle ();
       }
   }
+
+  /* log_rpl_calculate_replication_delay - calculate delay based on a given start time value
+   *        and the current time and log to the perfmon infrastructure; all calculations are
+   *        done in milliseconds as that is the relevant scale needed
+   *
+   *  a_start_time_source: the name of the source where this
+   */
+  int
+  log_rpl_calculate_replication_delay (THREAD_ENTRY *thread_p, time_t a_start_time_msec,
+				       const char *a_start_time_source)
+  {
+    // skip calculation if bogus input (sometimes, it is -1);
+    // TODO: fix bogus input at the source if at all possible
+    if (a_start_time_msec > 0)
+      {
+	const int64_t end_time_msec = util_gettime_msec ();
+	const double time_diff_msec = difftime (end_time_msec, a_start_time_msec);
+	assert (time_diff_msec > .0);
+	perfmon_set_stat (thread_p, PSTAT_SC_REPL_DELAY, time_diff_msec, PSTAT_SC_REPL_DELAY_ALWAYS_COLLECT);
+
+#if !defined(NDEBUG)
+	const double time_diff_sec = time_diff_msec / 1000.0;
+	er_log_debug (ARG_FILE_LINE, "repl_delay(%s): %9g sec, %9.2f msec",
+		      a_start_time_source, time_diff_sec, time_diff_msec);

```

Comment:
Suggestion for a separate patch: add a system parameter for replication error logging.

Reply:
Done